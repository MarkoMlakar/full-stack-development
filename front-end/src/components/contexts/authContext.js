import { createContext } from "react";

export const AuthContext = createContext({
  first_name: "",
  last_name: "",
  valid: false,
  id: "",
  dob: "",
  sightings: 0,
  email: "",
});
