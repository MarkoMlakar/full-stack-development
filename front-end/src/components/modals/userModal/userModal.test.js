import React from "react";
import { render, waitFor, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import UserModal from "./userModal";
import { AuthContext } from "../../contexts/authContext";
import axiosMock from "./__mocks__/axiosMock";
import { act } from "react-dom/test-utils";

let container;

// We need to create a container in which the modal will render
let modalRoot = document.getElementById("modal");
if (!modalRoot) {
  modalRoot = document.createElement("div");
  modalRoot.setAttribute("id", "modal");
  document.body.appendChild(modalRoot);
}

beforeEach(() => {
  container = document.createElement("modal");
  document.body.appendChild(container);
});

afterEach(() => {
  document.body.removeChild(container);
  container = null;
});

// Mock useContext
const mockUserContext = {
  user: {
    id: 1,
    first_name: "Marko",
    last_name: "Mlakar",
    dob: "April 3, 1996",
    email: "markomlakar2@gmail.com",
    sightings: 12,
  },
};
test("gets user information", async () => {
  //--> in case we want to call the api in userModal
  //   // Mock the axios API call
  //   // The actual /me endpoint doesn't actually
  //   axiosMock.get.mockResolvedValueOnce({
  //     user: {
  //   id: 1,
  //   first_name: "Marko",
  //   last_name: "Mlakar",
  //   dob: "April 3, 1996",
  //   email: "markomlakar2@gmail.com",
  //     },
  //   });

  const { getByTestId } = render(
    <AuthContext.Provider value={mockUserContext}>
      <UserModal showModal={true} />
    </AuthContext.Provider>
  );

  const firstName = await waitFor(() => getByTestId("first-name"));
  const lastName = await waitFor(() => getByTestId("last-name"));
  const dob = await waitFor(() => getByTestId("dob"));
  const email = await waitFor(() => getByTestId("email"));
  const sightings = await waitFor(() => getByTestId("sightings"));

  expect(firstName).toHaveTextContent("Marko");
  expect(lastName).toHaveTextContent("Mlakar");
  expect(dob).toHaveTextContent("April 3, 1996");
  expect(email).toHaveTextContent("markomlakar2@gmail.com");
  expect(sightings).toHaveTextContent("12 sightings");
});

test("renders correctly", () => {
  const tree = render(
    <AuthContext.Provider value={mockUserContext}>
      <UserModal showModal={true} />
    </AuthContext.Provider>
  );
  expect(tree).toMatchSnapshot();
});
