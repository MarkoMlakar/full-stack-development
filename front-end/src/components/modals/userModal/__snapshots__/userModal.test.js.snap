// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      id="modal"
    >
      <div
        class="overlay"
      />
      <div
        class="modal-container"
      >
        <div
          class="user-modal"
        >
          <div
            class="close-btn"
          >
            <img
              alt="Close"
              class="close-image"
              src="close-icon.svg"
            />
          </div>
          <div
            class="user-info-container"
          >
            <div
              class="user-header"
            >
              <img
                alt="User"
                class="user-image"
                src="profile-image.png"
              />
              <div
                class="user-header-name"
              >
                <div
                  class="user-name-title"
                >
                  Marko
                  Â 
                  Mlakar
                </div>
                <div
                  class="user-sightings-title"
                  data-testid="sightings"
                >
                  12
                   sightings
                </div>
              </div>
            </div>
            <div
              class="user-info"
            >
              <div
                class="user-info-row"
              >
                <div
                  class="user-row-title"
                >
                  First Name
                </div>
                <div
                  class="user-row-value"
                  data-testid="first-name"
                >
                  Marko
                </div>
              </div>
              <div
                class="user-info-row"
              >
                <div
                  class="user-row-title"
                >
                  Last Name
                </div>
                <div
                  class="user-row-value"
                  data-testid="last-name"
                >
                  Mlakar
                </div>
              </div>
              <div
                class="user-info-row"
              >
                <div
                  class="user-row-title"
                >
                  Date of Birth
                </div>
                <div
                  class="user-row-value"
                  data-testid="dob"
                >
                  April 3, 1996
                </div>
              </div>
              <div
                class="user-info-row"
              >
                <div
                  class="user-row-title"
                >
                  Email
                </div>
                <div
                  class="user-row-value"
                  data-testid="email"
                >
                  markomlakar2@gmail.com
                </div>
              </div>
            </div>
          </div>
          <div
            class="user-logout"
          >
            <button
              class="logout-button"
            >
              <div
                class="logout-label"
              >
                Logout
              </div>
            </button>
          </div>
        </div>
      </div>
    </div>
    <modal />
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
